src/components/utils/window.js

```diff
@@ -0,0 +1,3 @@
+const windowCopy = window
```

Comment:
I don't understand this `copy` operator. By default, there're no copies performed with `=` operator. For shallow copying, you can use the spread operator:
```suggestion
const windowCopy = { ...window }
```
But I want to hear more about the reasons behind.

Reply:
Yep, I meant `const windowCopy = { ...window }`. Anyways, decided to minimise the changes and just use `window` directly in `safeWoopra.js`